src/spec.c

```diff
@@ -1063,6 +1033,10 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, StrongRef spec_ref, ArgsCu
   IndexSpecCache_Decref(sp->spcache);
   sp->spcache = IndexSpec_BuildSpecCache(sp);
 
+  for (size_t ii = prevNumFields; ii < sp->numFields; ++ii) {
+    FieldsGlobalStats_UpdateStats(sp->fields + ii, 1);
+  }
```

Comment:
Please explain why it moved here?

Reply:
This function is used to add new fields, both by `FT.CREATE` and `FT.ALTER`. It makes sense to update the statistics here when we know we added them correctly. We also know we hold the GIL at this moment.
The complementary function was moved into `IndexSpec_RemoveFromGlobals`, where we mark the spec as deleted and make it unreachable from the outside, and we also know we have the GIL there